src/unread/unreadStreamsReducer.js

```diff
@@ -16,11 +16,11 @@ import { getOwnUserId } from '../users/userSelectors';
 const initialState: UnreadStreamsState = NULL_ARRAY;
 
 const eventNewMessage = (state, action, globalState) => {
-  if (action.message.type !== 'stream') {
+  if (getOwnUserId(globalState) === action.message.sender_id) {
     return state;
   }
 
-  if (getOwnUserId(globalState) === action.message.sender_id) {
+  if (action.message.type !== 'stream') {
     return state;
```

Comment:
- When we start putting things together to call
      `addItemsToStreamArray`, we need to be sure `action.message` is a
      stream message. We were already sure of that; there's already an
      early return on `action.message.type !== stream`. But Flow seems
      to think it's done too early, and that something in the
      intervening code might thwart the job it's doing. So, move that
      early return a bit later.

Ah, yep. Specifically what's happening is: we check `action.message.type`, but then we call `getOwnUserId` -- and that's some function that might go and do anything. So in particular it might go and mutate the same object that we're looking at as `action.message`. So that object *had* had a `type` of `'private'`, but now it might not and so it might not have a `stream_id`.